app/uk/gov/hmrc/agentservicesaccount/connectors/AgentClientAuthorisationConnector.scala

```diff
@@ -46,4 +46,15 @@ class AgentClientAuthorisationConnector @Inject()(http: HttpClient)(implicit val
             case NOT_FOUND => throw SuspensionDetailsNotFound("No record found for this agent")
           })
     }
+
+  def getAgencyDetails()(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[AgencyDetails]] =
```

Comment:
Sorry, I missed this in the backend service, but AgentServicesController should be returning a 404 not a 204 when the resource (agency-details) can't be found.

Reply:
merging this so everything is consistent, but we should change that status code in another PR